cluster/lib/redis/cluster/client.rb

```diff
@@ -54,11 +54,11 @@ def initialize(*)
       ruby2_keywords :initialize if respond_to?(:ruby2_keywords, true)
 
       def id
-        @router.node_keys.join(' ')
+        server_url.join(' ')
       end
 
       def server_url
-        @router.node_keys
+        @router.nil? ? @config.startup_nodes.keys : router.node_keys
```

Comment:
Hi! 👋

This is related to a question I asked in https://github.com/redis-rb/redis-cluster-client/pull/364/files#r1724410379 . 

I noticed that here too, the `@router` variable is being used directly instead of calling the `router` method. What's the reasoning behind this approach?

It seems that calling `router` would eliminate the need for the nil check. Is there a specific advantage to using `@router` directly in this context?

Also, I am waiting for this fix to be released!

Thanks in advance for your help!

Reply:
I thought that it would be better to delay initialization if any commands aren't sent to servers in the method.